Create BoardAPI

marcinogo opened the pull request.

marcin-ogorzalek-EPAM committed:
Write progress estimates for 14.03.2019

marcin-ogorzalek-EPAM committed:
Add dependency for TestNG

marcin-ogorzalek-EPAM committed:
Update section AboutOX

marcin-ogorzalek-EPAM committed:
Create .gitattributes

marcin-ogorzalek-EPAM committed:
Create .mailmap file

marcin-ogorzalek-EPAM committed:
Update estymaty.md for 14.03.2019

marcin-ogorzalek-EPAM committed:
Change to proper description of project

marcin-ogorzalek-EPAM committed:
Create flow diagrams for application

marcin-ogorzalek-EPAM committed:
Create CRC cards for application

marcin-ogorzalek-EPAM committed:
Create flowchart for game and setup of game

marcinogo committed:
Redraw flowchart for the project

Due to low quality, long time of loading jpg file live on github and
mistakes flowcharts had been redrawn using Dia appliaction.

marcinogo committed:
Add new process to setup flowchart

marcinogo committed:
Redraw CRC cards

Due to low qulity and mistakes CRC radds has been redrawn.

marcin-ogorzalek-EPAM committed:
Merge branch 'master' of github.com:marcinogo/OX into dev

marcin-ogorzalek-EPAM committed:
Merge branch 'dev' of github.com:marcinogo/OX into dev

marcin-ogorzalek-EPAM committed:
Delete wrong flowchart

marcin-ogorzalek-EPAM committed:
Rename CRC cards file

marcin-ogorzalek-EPAM committed:
Update estymaty.md for 19.03

marcin-ogorzalek-EPAM committed:
Create cass Field with field sign, create Sign enum

marcin-ogorzalek-EPAM committed:
Implement method equals()

marcin-ogorzalek-EPAM committed:
Add O and X constants in Sign enum

marcin-ogorzalek-EPAM committed:
Implement hashCode()

marcin-ogorzalek-EPAM committed:
Refractor tests

Remove test duplication, update messages for failed assertation, add
additional test cases.

marcin-ogorzalek-EPAM committed:
Implement changeSign()

marcin-ogorzalek-EPAM committed:
Implement toString()

marcin-ogorzalek-EPAM committed:
Refractor testIfToStringReturnCorrectString

marcin-ogorzalek-EPAM committed:
Reduce visibility of Field class

marcin-ogorzalek-EPAM committed:
Create Board class with fields height and width

marcin-ogorzalek-EPAM committed:
Refractor testIfBoardHaveDimensions()

marcin-ogorzalek-EPAM committed:
Add validation for proper dimensions at Board creation

marcin-ogorzalek-EPAM committed:
Add field matirx in Boardi

marcin-ogorzalek-EPAM committed:
Change upper bound for test cases

Due to OutOfMemoryError with array of length Integer.MAX_VALUE upper
bound had to be change. Additionaly for redibilyt of board max value has
been choose as 200.

marcin-ogorzalek-EPAM committed:
Add validation for Board dimensions upper bound

marcin-ogorzalek-EPAM committed:
Refractor tests for check validation of improper dimesnions

marcin-ogorzalek-EPAM committed:
Add width to field matrix

marcin-ogorzalek-EPAM committed:
Implement setField()

marcin-ogorzalek-EPAM committed:
Refractor testIfBoardSetFieldValueInMatrix()

marcin-ogorzalek-EPAM committed:
Fix copy constructor, implemnt toString()

Due to logical error matrix field of created copy had been reset every
time on it creation.
Additional to this proper String representation of Board had been
implemented.

marcin-ogorzalek-EPAM committed:
Refractor testIfToStringReturnCorrectString

marcin-ogorzalek-EPAM committed:
Create BoardFactory with createBoard() and implement it

marcin-ogorzalek-EPAM committed:
Create Main class with main() method

marcinogo committed:
Move classes to package board

marcinogo committed:
Correct visibility in class Board

marcinogo committed:
Update Baord with new upper bound

Upper bound for Board width and height had been set as 40.

marcinogo committed:
Move helper method from test to Board

marcinogo committed:
Update logic with use setBoardMatrixCells()

marcinogo committed:
Create test print of empty Board

marcinogo committed:
Create script to run application

marcinogo committed:
Creat BoardAPI and implement it

marcin-ogorzalek-EPAM committed:
Implement equals()

marcin-ogorzalek-EPAM committed:
Implement hashCode()

marcin-ogorzalek-EPAM committed:
Add test for createBoard()

marcin-ogorzalek-EPAM committed:
Test setBoardMatrixCells() method

marcin-ogorzalek-EPAM committed:
Update estymaty.md for 20.03

marcin-ogorzalek-EPAM committed:
Fix bug with copy matrix content

marcin-ogorzalek-EPAM committed:
Test testSetField()

marcinogo committed:
Create and implement PlayerAPI (#13)

* Create Player class

* Implement validation for player name

To create player any name must be given. For this purpose validation had
been implemented. Player name can not be null.

* Update player name validation

* Implement sign validation for null case

* Update sign validation for default case

* Add field points

* Implement equals()

* Implement hashCode()

* Add constructor with parameter points

There can be possibilty that we want to creat player with given number
of points. More to that we have to be sure that player have at least 0
or more points.

* Create setPlayerPoints() method

* Override toString()

* Create PlayerFactory

* Fix test cases

* Move classes to proper packages

* Reduce visibility of PlayerFactory

* Create and implement PlayerAPI

* Test PlayerAPIImpl

marcin-ogorzalek-EPAM committed:
Change player class visibility

WRup commented on src/test/java/ogo/marcin/ox/BoardTest.java:
```diff
@@ -77,29 +77,28 @@ public void testIfBoardHaveMatrixOfGivenWidth(int width, int height) {
     public void testIfBoardSetFieldValueInMatrix(int width, int height,
                                                  int widthToUpdate, int heightToUpdate,
-                                                 Sign fieldSign) {
+                                                 Field field) {
```
What about changeState() method to change state of Field?

WRup commented on src/test/java/ogo/marcin/ox/board/BoardAPIImplTest.java:
```diff
@@ -0,0 +1,66 @@
+        Board result = boardAPI.setField(defaultBoard, widthToUpdate, heightToUpdate, newSign);
+        assert !defaultBoard.equals(result) : "Boards should not be equals";
+        defaultBoard.matrix[heightToUpdate][widthToUpdate] = new Field(newSign);
+        assert defaultBoard.equals(result) : "Boards should be equals";
```
Mby one asser per one Test?

WRup commented:
Please check my comments, review and push to master.

marcin-ogorzalek-EPAM committed:
Update setField() in Board to use dedicated method of Field

marcin-ogorzalek-EPAM committed:
Create additional tests

To keep assertations at 1 per test, I created new tests for Factory and
API classes.
